app/models/vm_or_template.rb

```diff
@@ -1193,7 +1193,7 @@ def refresh_on_reconfig
     unless ext_management_system.authentication_status_ok?
       raise _("%{table} failed last authentication check") % {:table => ui_lookup(:table => "ext_management_systems")}
     end
-    EmsRefresh.reconfig_refresh(self)
+    EmsRefresh.refresh(self)
```

Comment:
It seems the important point here is that the call is synchronous so when the `reconfigvm_task_complete` method runs it updates the inventory first before making the policy call to `vm_reconfigure`.

[db/fixtures/ae_datastore/ManageIQ/System/Event/EmsEvent/EVM.class/reconfigvm_task_complete.yaml#L11](https://github.com/ManageIQ/manageiq/blob/master/db/fixtures/ae_datastore/ManageIQ/System/Event/EmsEvent/EVM.class/reconfigvm_task_complete.yaml#L11)